### ISSUE TITLE
Integrate mypy

### ISSUE THREAD

**Issue by sobolevn** [lk-geimfari/mimesis#263]
When #258 will be ready the next step is to integrate `mypy`.

There is a standalone script.
And there's a plugin for `pytest`, but I have not used it yet. https://pypi.python.org/pypi/pytest-mypy

**Comment by lk-geimfari** [comment]
@sobolevn Unfortunately, the solution works slowly... too slowly: `3m 42s 840ms`

**Comment by lk-geimfari** [comment]
Also it's doesn't support `mypy.ini` file. It's a problem.

**Comment by lk-geimfari** [comment]
I think that i'll rewrite plugin especially for us.

**Comment by sobolevn** [comment]
@lk-geimfari that plugin is not production ready, that's true. But maybe should help with its development instead of rewriting it?

**Comment by lk-geimfari** [comment]
@sobolevn Of course, I'll write to the author and if he agrees, I'll work on current plugin. Rewriting is an extreme measure.

**Comment by lk-geimfari** [comment]
@sobolevn How should we check types? Should we check types with usual tests or after ones? 

Let me illustrate ideas:

```bash
py.test --verbose --color=yes --mypy mimesis
```
or
```bash
make test # 
make type-check #
```

The problem with co-testing is that mypy takes a considerable amount of time.

**Comment by sobolevn** [comment]
What's the timings for tests and type checking?

**Comment by lk-geimfari** [comment]
@sobolevn I have run `py.test` 7 minutes ago and testing still going testing.

**Comment by sobolevn** [comment]
`mypy` must be way faster. So, it is easier to run fast tests first, then slow tests.
I will also open a new issue to optimize tests speed.

**Comment by lk-geimfari** [comment]
Fixed in #299
